src/Components/ListCategories.jsx

```diff
@@ -6,28 +6,39 @@ class ListCategories extends Component {
     super(props);
     this.state = {
       categories: [],
+      gotCategories: false,
     };
   }
 
   componentDidMount() {
-    this.HandleListCategories();
+    const { gotCategories } = this.state;
+    if (!gotCategories) this.handleListCategories();
```

Comment:
Este if parece também ser redundante.

```suggestion
    this.handleListCategories();
```

Reply:
Excelente trabalho, Jonnes! :100:

Reply:
Alterações feitas ✅